File: 김남형/Week4/BOJ_1138_한 줄로 서기.py
n = int(input())
arr = list(map(int, input().split()))
ans = [0] * n

for i in range(n): # 
    cnt = 0 # 빈 자리 카운트
    for j in range(n):
        if cnt == arr[i] and ans[j] == 0: # 본인 인덱스에 주어진 수와 앞에 빈 자리 수가 같고, 현재 자리가 비었으면
            ans[j] = i+1
            break
        elif ans[j] == 0: # 자리가 비어있으면 cnt 증가
            cnt += 1
        
print(*ans)

# 본인 인덱스에 주어진 수만큼 앞에 자리를 비워두고 빈 자리에 찾아가면 된다.

File: 김한비/Week4/BOJ_1138_한 줄로 서기.cpp
#include <iostream>

using namespace std;

int arr[10];

int main() {

	int n;
	cin >> n;
	for (int i = 1; i <= n; i++) {
		int cnt = 0;
		cin >> cnt;

		for (int j = 0; j < n; j++) {
			if (cnt == 0 && arr[j] == 0) {
				arr[j] = i;
				break;
			}
			if (arr[j] == 0)	cnt--;
		}
	}

	for (int i = 0; i < n; i++)
		cout << arr[i] << " ";

	return 0;
}
